forth_src/io.fs

```diff
@@ -0,0 +1,75 @@
+require open
+
+\ select a file for input
+\ no error handling
+code (chkin) ( file# -- result )
+w stx,
+lsb lda,x tax, \ x = file#
+$ffc6 jsr, \ CHKIN
+0 lda,#
++branch bcc, \ carry clear: OK
+1 lda,# \ carry set: error
+:+
+w ldx, lsb sta,x \ push result
+;code
+
+\ Set output device to open file #
+code (chkout) ( file# -- result )
+w stx,
+lsb lda,x tax, \ x = file#
+$ffc9 jsr, \ CHKOUT
+0 lda,#
++branch bcc, \ carry clear: OK
+1 lda,# \ carry set: error
+:+
+w ldx, lsb sta,x \ push result
+;code
+
+\ Reset input and output to console 
+code clrch ( -- ) 
+txa, pha,
+$ffcc jsr,  \ CLRCH
+pla, tax,
+;code
+
+\ Read status of last IO operation
+code readst ( -- status )
+dex, 0 lda,# msb sta,x
+$ffb7 jsr, \ READST
+lsb sta,x
+;code
+
+\ Get a byte from input device
+code chrin ( -- chr )
+dex, w stx, 0 lda,# msb sta,x
+$ffcf jsr, \ CHRIN
+w ldx, lsb sta,x
+;code
+
+\ handle returned result value
+: ioerr ( file# result -- )
+if clrch close rvs ." io err" cr abort 
```

Comment:
That `ioresult` idea is great; definitely makes things a bit cleaner.

Reading the error channel would definitely be more user-friendly, though correctly handling cases like a nonexistent device, or devices that aren't disk drives, will add a bit more complexity. My intent with `ioerr` was to just be a minimal safety net, but if you think the extra complexity is warranted, I'll add it in. I suppose we could expose `_errorchread` from disk.asm as a word.

Reply:
I didn't think about all those special cases. I think it might be nice for `dos` and `send-cmd` since those are specifically intended for use with drives. Not for `open`/`close` which are supposed to work with anything.

Reply:
`dos` and `send-cmd` already prints the error channel after sending a command (and can be used to just read the error channel if you send an empty command), I agree with you about `open`.

I did find out that the kernal routines for `open`, `chkin` and `chkout` leave an error number in A when they fail, so I've reworked those words to capture that error number, and given `ioerr` some more meaningful error messages.